packages/gestalt/src/RadioGroupButton.js

```diff
@@ -0,0 +1,192 @@
+   * Ref forwarded to the underlying input element. See [ref example](https://gestalt.pinterest.systems/radiogroup#Using-ref) for more details.
+   */
+  ref?: HTMLInputElement, // eslint-disable-line react/no-unused-prop-types
+  /**
+   * sm: 16px, md: 24px
+   */
+  size?: 'sm' | 'md',
+  /**
+   * Optional description for the input, used to provide more detail about an option. See the [subtext example](https://gestalt.pinterest.systems/radiogroup#With-subtext) for more details.
+   */
+  subtext?: string,
+  /**
+   * The value of the input.
+   */
+  value: string,
+|};
+
+/**
+ *  Use [RadioGroup.RadioButtons](https://gestalt.pinterest.systems/radiogroup#RadioGroup.RadioButton) to present an option for selection to the user within a RadioGroup. They should not be used outside of RadioGroup or when the user can select more than one option from a list.
+ *
+ * ![RadioButton light mode](https://raw.githubusercontent.com/pinterest/gestalt/master/cypress/integration/visual-test/__image_snapshots__/RadioButton%20%230.png)
+ * ![RadioButton dark mode](https://raw.githubusercontent.com/pinterest/gestalt/master/cypress/integration/visual-test/__image_snapshots__/RadioButton-dark%20%230.png)
+ *
+ */
+const RadioGroupButtonWithForwardRef: React$AbstractComponent<Props, HTMLInputElement> = forwardRef<
+  Props,
+  HTMLInputElement,
+>(function RadioButton(
+  {
+    checked = false,
+    disabled = false,
+    id,
+    image,
+    label,
+    name,
+    onChange,
+    subtext,
+    value,
+    size = 'md',
+  }: Props,
+  ref,
+): Node {
+  const [focused, setFocused] = useState(false);
+  const [hovered, setHover] = useState(false);
+
+  const handleChange: (event: SyntheticInputEvent<HTMLInputElement>) => mixed = (event) =>
+    onChange({ checked: event.target.checked, event });
+
+  const handleBlur: () => void = () => setFocused(false);
+
+  const handleFocus: () => void = () => setFocused(true);
+
+  const handleHover: (isHovered: boolean) => void = (isHovered: boolean) => setHover(isHovered);
+
+  let borderStyle = styles.Border;
+  if (disabled && checked) {
+    borderStyle = styles.BorderDisabledChecked;
+  } else if (!disabled && checked) {
+    borderStyle = styles.BorderSelected;
+  } else if (!disabled && hovered) {
+    borderStyle = styles.BorderHovered;
+  }
+
+  let borderWidth = styles.BorderUnchecked;
+  if (disabled && !checked) {
+    borderWidth = styles.BorderDisabled;
+  } else if (checked && size === 'sm') {
+    borderWidth = styles.BorderCheckedSm;
+  } else if (checked && size === 'md') {
+    borderWidth = styles.BorderCheckedMd;
+  }
+
+  const styleSize = size === 'sm' ? controlStyles.sizeSm : controlStyles.sizeMd;
+
+  const bgStyle = disabled && !checked ? styles.BgDisabled : styles.BgEnabled;
+
+  const { isFocusVisible } = useFocusVisible();
+
+  const { parentName } = useRadioGroupContext();
+  if (parentName !== 'RadioGroup') {
+    throw new Error(`RadioGroup.RadioButton must be used within a RadioGroup.`);
```

Comment:
can we add link here? 

    throw new Error(`RadioGroup.RadioButton must be used within a [RadioGroup](https://gestalt.pinterest.systems/radiogroup).`);